frontend/public/components/routes/RoutePage.tsx

```diff
@@ -0,0 +1,146 @@
+import { safeJSToYAML, safeYAMLToJS } from '@console/shared/src/utils/yaml';
+import { baseTemplates } from '../../models/yaml-templates';
+import { RouteForm } from './RouteForm';
+import { useTranslation } from 'react-i18next';
+import {
+  convertEditFormToRoute,
+  convertRouteToEditForm,
+  createRouteUpdatePatches,
+  routeValidationSchema,
+} from './utils';
+import { RouteFormProps } from './create-route';
+
+type RouteFormValues = {
+  editorType: EditorType;
+  formData: RouteFormProps;
+  yamlData: string;
+};
+
+const defaultRouteYAML = baseTemplates.get(referenceForModel(RouteModel)).get('default');
+
+export type RoutePageProps = RouteComponentProps<{ ns?: string; name?: string }>;
+
+export const RoutePage: React.FC<RoutePageProps> = ({ match }) => {
+  const { t } = useTranslation();
+  const namespace = match.params.ns;
+  const name = match.params.name;
+  const isEditForm = !!name;
+  const heading = isEditForm ? t('public~Edit Route') : t('public~Create Route');
+  const submitLabel = isEditForm ? t('public~Save') : t('public~Create');
+
+  const [route, routeLoaded, routeLoadError] = useK8sWatchResource<RouteKind>({
+    kind: RouteModel.kind,
+    name,
+    namespace,
+  });
+  const [services, servicesLoaded, serviceLoadError] = useK8sWatchResource<K8sResourceKind[]>({
+    kind: ServiceModel.kind,
+    namespace,
+    isList: true,
+  });
+  const loaded = routeLoaded && servicesLoaded;
+
+  const canUpdateHost = useAccessReviewAllowed({
+    group: RouteModel.apiGroup,
+    resource: 'routes/custom-host',
+    verb: 'update',
+    name,
+    namespace,
+  });
+
+  const initialValues = React.useMemo(
+    () => ({
+      editorType: EditorType.Form,
+      yamlData: isEditForm
+        ? safeJSToYAML(route, 'yamlData', {
+            skipInvalid: true,
+          })
+        : defaultRouteYAML,
+      formData: isEditForm ? convertRouteToEditForm(services, route) : {},
+    }),
+    [isEditForm, route, services],
+  );
+
+  const handleSubmit = async (values: RouteFormValues, helpers: FormikHelpers<RouteFormValues>) => {
+    const data: RouteKind =
+      values.editorType === EditorType.Form
+        ? convertEditFormToRoute(values.formData, isEditForm && route)
+        : safeYAMLToJS(values.yamlData);
+
+    if (!canUpdateHost && isEditForm && data.spec.host !== route.spec.host) {
+      helpers.setStatus({
+        submitSuccess: '',
+        submitError: t('public~Insufficient permissions to update host.'),
+      });
+      return null;
+    }
+
+    let resource: RouteKind;
+    try {
+      if (isEditForm) {
+        if (values.editorType === EditorType.Form) {
+          resource = await k8sPatchResource({
+            model: RouteModel,
+            resource: route,
+            data: createRouteUpdatePatches(values.formData, initialValues.formData, route),
+          });
+        } else {
+          resource = await k8sUpdateResource({ model: RouteModel, data, name });
+        }
```

Comment:
in edit flow for form view, we are making patch `k8sPatchResource` and for YAML we have an update `k8sUpdateResource`. I know it's not straightforward to get the data for the patch in the case of the YAML view but ideally for both flows same call should be made I think and if we can't patch for both should we stick with an update call here for both views

@spadgett let us know your thoughts?

cc @rottencandy

Reply:
I'm fine with update.